pandas/core/arrays/timedeltas.py

```diff
@@ -997,6 +989,30 @@ def objects_to_td64ns(data, unit="ns", errors="raise"):
     return result.view('timedelta64[ns]')
 
 
+def _validate_td64_dtype(dtype):
+    try:
+        if dtype == np.dtype("timedelta64"):
+            dtype = _TD_DTYPE
+            msg = textwrap.dedent("""\
+                Passing in 'timedelta' dtype with no precision is deprecated
```

Comment:
why are you warning? this *already* raises
```
In [5]: pd.TimedeltaIndex([1,2,3],dtype='timedelta64')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-65a1792bd679> in <module>()
----> 1 pd.TimedeltaIndex([1,2,3],dtype='timedelta64')

~/pandas-dev/pandas/core/indexes/timedeltas.py in __new__(cls, data, unit, freq, start, end, periods, closed, dtype, copy, name, verify_integrity)
    224 
    225         tdarr = TimedeltaArray._from_sequence(data, freq=freq, unit=unit,
--> 226                                               dtype=dtype, copy=copy)
    227         return cls._simple_new(tdarr._data, freq=tdarr.freq, name=name)
    228 

~/pandas-dev/pandas/core/arrays/timedeltas.py in _from_sequence(cls, data, dtype, copy, freq, unit)
    206                        freq=None, unit=None):
    207         if dtype != _TD_DTYPE:
--> 208             raise ValueError("Only timedelta64[ns] dtype is valid.")
    209 
    210         freq, freq_infer = dtl.maybe_infer_freq(freq)

ValueError: Only timedelta64[ns] dtype is valid.
```

Reply:
Did you see this issue? This didn't raise on 0.23.4.